src/Main.java

```diff
@@ -1,278 +1 @@
-    }
-
-    static int GetCellwithLowestCost(List<Cell> path){
-        int cost = Integer.MAX_VALUE;
-        int posn = 0;
-        for (int i = 0; i < path.size(); i++){
-            if (path.get(i).gCost < cost) {
-                cost = path.get(i).gCost;
-                posn = i;
-            }
-        }
-        return posn;
-    }
-
-    static Cell GetCellfromList(List<Cell> path, Cell cell){
-        for (int i = 0; i < path.size(); i++){
-            if (CompareCells(path.get(i), cell)) return path.get(i);
-        }
-        return null;
-    }
-
-    static Boolean isCellinList(List<Cell> path, Cell cell){
-        for (int i = 0; i < path.size(); i++){
-            if (CompareCells(path.get(i), cell)) return true;
-        }
-        return false;
-    }
-
-    static Boolean CompareCells(Cell a, Cell b){
-        if (a.x == b.x && a.y == b.y) return true;
-        else return false;
-    }
-
-    static Boolean isValidMove(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (cell.x < 0 || cell.x >= m || cell.y < 0 || cell.y >= n || maze[cell.x][cell.y] == '#'
-                || maze[cell.x][cell.y] == 'X') return false;
-        return true;
-    }
-
-    static Boolean isValidMoveAStar(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (cell.x < 0 || cell.x >= m || cell.y < 0 || cell.y >= n || maze[cell.x][cell.y] == '#') return false;
-        return true;
-    }
-}
-
-public class Main {
-    public static void main(String[] args) {
-        // write your code here
-
-        try {
-            // Specify input path here
-            Scanner sc = new Scanner(new File("C:\\Users\\jimfa\\Desktop\\mazer\\mazer\\test.txt"));
-
-            // Comment this out if you do not want to output to file
-            PrintStream output = new PrintStream(new File("Solved_Maze.txt"));
-            System.setOut(output);
-
-            // Control which kind of algorithm you want to use, please turn only ONE on
-            Boolean useDFS = false;
-            Boolean useBFS = true;
-            Boolean useAStar = false;
-
-            int m = sc.nextInt();
-            int n = sc.nextInt();
-            int startx = sc.nextInt();
-            int starty = sc.nextInt();
-            Cell cell = new Cell(startx, starty, null, 0, 0);
-            int endx = sc.nextInt();
-            int endy = sc.nextInt();
-            Cell goal = new Cell (endx, endy, null, 0,0);
-            char[][] maze = new char[m][n];
-
-            for (int i = 0; i < m; i++) {
-                for (int j = 0; j < n; j++) {
-                    int temp = sc.nextInt();
-                    if (temp == 1) {
-                        maze[i][j] = '#';
-                    } else if (temp == 0) {
-                        maze[i][j] = ' ';
-                    }
-                    if (i == startx && j == starty) {
-                        maze[i][j] = 'S';
```

Comment:
Any reason to already set it here if we need to re-set it below? I mean that for the DFS, we need to re-set the maze start point again. Maybe we shouldn't need to do that? :)